client/state/data-layer/wpcom/read/tags/mine/delete/test/index.js

```diff
@@ -0,0 +1,135 @@
+/*
+ * External dependencies
+ */
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import {
+	requestUnfollowTag as requestUnfollowAction,
+	receiveUnfollowTag as receiveUnfollowAction,
+} from 'state/reader/tags/items/actions';
+import {
+	requestUnfollow,
+	receiveUnfollowTag,
+	receiveError,
+	fromApi,
+} from '../';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { NOTICE_CREATE } from 'state/action-types';
+
+const successfulUnfollowResponse = {
+	subscribed: false,
+	removed_tag: '307',
+	tags: [
+		{
+			ID: '422',
+			slug: 'poetry',
+			title: 'Poetry',
+			display_name: 'poetry',
+			URL: 'https://public-api.wordpress.com/rest/v1/read/tags/poetry/posts'
+		},
+		{
+			ID: '69750',
+			slug: 'ship',
+			title: 'SHIP',
+			display_name: 'ship',
+			URL: 'https://public-api.wordpress.com/rest/v1/read/tags/ship/posts'
+		},
+	],
+};
+
+const unsuccessfulResponse = {
+	...successfulUnfollowResponse,
+	subscribed: true,
+};
```

Comment:
I'm pretty sure 400s/500s get automatically redirected to the error response so I don't think I need to explicitly handle those like I do a sneaky 200 failure